File: Selenium/Activities/Activity21.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as fireFoxService
from selenium.webdriver.common.by import By
import time


service=fireFoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service)as driver:
    driver.get("https://www.training-support.net/selenium/tab-opener")
    print("Page Title :- " + driver.title)
    driver.find_element(By.ID,"launcher").click()
    window=driver.window_handles
    itr=window.__iter__()
    id=[]

    for win in itr:
        id.append(win)

    driver.switch_to.window(id[len(id)-1])
    time.sleep(2)
    print("New Window title :- " + driver.title)
    header=driver.find_element(By.CLASS_NAME,"content").text
    print("First Header :- " + header)

    driver.find_element(By.ID,"actionButton").click()

    window=driver.window_handles
    itr2=window.__iter__()
    for win in itr2:
        id.append(win)
    driver.switch_to.window(id[len(id)-1])
    time.sleep(2)
    print("New Window title :- " + driver.title)
    header2=driver.find_element(By.CLASS_NAME,"content").text
    print("Second Header :- " +header2)

    driver.quit()



    
    

File: Selenium/Activities/Activity21.java
package example;

import java.util.Set;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;

public class Activity21 {

	public static void main(String[] args) {
		WebDriver driver = new ChromeDriver();

		driver.get("https://training-support.net/webelements/tabs");
		System.out.println("Parent Title of the page is: " + driver.getTitle());

		String parentWindow = driver.getWindowHandle();
		System.out.println("parentWindow id: " + parentWindow);


		driver.findElement(By.xpath("//button[text()='Open A New Tab']")).click();

		// Switch to first child
		String firstChild = null;

		for (String window : driver.getWindowHandles()) {
		    if (!window.equals(parentWindow)) {
		        firstChild = window;
		        driver.switchTo().window(firstChild);
		        break;
		    }
		}

		System.out.println("First child title: " + driver.getTitle());
		System.out.println("First child window id: " + firstChild);

		// Store windows before opening third window
		Set<String> oldWindows = driver.getWindowHandles();

		driver.findElement(By.xpath("//button[text()='Open Another One']")).click();

		// Get third window
		String thirdWindow = null;

		for (String window : driver.getWindowHandles()) {
		    if (!oldWindows.contains(window)) {
		        thirdWindow = window;
		        driver.switchTo().window(thirdWindow);
		        break;
		    }
		}

		System.out.println("Third window title: " + driver.getTitle());
		System.out.println("Third window id: " + thirdWindow);
		
		driver.quit();
		
		
		
		
		
		
	}

}
